eox_nelp/signals/receivers.py

```diff
@@ -393,18 +393,26 @@ def pearson_vue_course_completion_handler(instance, **kwargs):  # pylint: disabl
     if not getattr(settings, "PEARSON_RTI_ACTIVATE_COMPLETION_GATE", False):
         return
 
-    if getattr(settings, "USE_PEARSON_ENGINE_SERVICE", False):
-        real_time_import_task_v2.delay(
-            user_id=instance.user_id,
-            exam_id=str(instance.context_key),
-            action_name="rti",
-        )
-    else:
-        real_time_import_task.delay(
-            user_id=instance.user_id,
-            course_id=str(instance.context_key),
+    is_complete, graded = get_completed_and_graded(user_id=instance.user_id, course_id=str(instance.context_key))
+
+    if is_complete and not graded:
+        LOGGER.info(
+            "Initializing rti task for the user %s, action triggered by course completion status",
+            instance.user_id,
         )
 
+        if getattr(settings, "USE_PEARSON_ENGINE_SERVICE", False):
+            real_time_import_task_v2.delay(
+                user_id=instance.user_id,
+                exam_id=str(instance.context_key),
+                action_name="rti",
+            )
+        else:
+            real_time_import_task.delay(
+                user_id=instance.user_id,
+                course_id=str(instance.context_key),
+            )
+
```

Comment:
```suggestion
    if (not is_complete) or graded:
        return
 
    LOGGER.info(
        "Initializing rti task for the user %s, action triggered by course completion status",
        instance.user_id,
    )

    if getattr(settings, "USE_PEARSON_ENGINE_SERVICE", False):
        real_time_import_task_v2.delay(
            user_id=instance.user_id,
            exam_id=str(instance.context_key),
            action_name="rti",
        )
    else:
        real_time_import_task.delay(
            user_id=instance.user_id,
            course_id=str(instance.context_key),
        )

```

Reply:
very similar to my first approach `if  graded or not is_complete:` but I decided to leave as similar as possible in order to make easier the review, anyway I will update that